openlibrary/plugins/openlibrary/code.py

```diff
@@ -360,23 +361,45 @@ def GET(self):
         raise web.HTTPError('200 OK', {}, 'OK')
 
 
+class isbn_lookup(delegate.page):
+
+    path = r'/isbn/([0-9xX-]+)'
+
+    def GET(self, isbn):
+        from openlibrary.core.models import Edition
+
+        # Preserve the url type (e.g. `.json`) and query params
+        ext = ''
+        if web.ctx.encoding and web.ctx.path.endswith('.' + web.ctx.encoding):
+            ext = '.' + web.ctx.encoding
+        if web.ctx.env.get('QUERY_STRING'):
+            ext += '?' + web.ctx.env['QUERY_STRING']
+
+        try:
+            ed = Edition.from_isbn(isbn)
+            if ed:
+                return web.found(ed.key + ext)
+        except Exception as e:
```

Comment:
add it to the linter rules

Reply:
The linter can not detect all things.  `make lint-diff` already detects and flags (flake8 E722) on the use of a true bare `except:`.  However, `except Exception as e:` is not flagged because it is assumed that the developer wants to sweep all exceptions under the carpet.

Note: flake8 E722 currently detects 74 instances in this repo.

Reply:
The main risk here is the `create_edition_from_amazon_metadata` function inside `from_isbn` which could throw (maybe?). The main thing that makes these things hard is that there's really no easy way to see what _could_ be thrown while looking at this code. It would be awesome if there was an IDE extension or something that looked at all the functions called, and gave you an idea of what _might_ be thrown. This is less of a problem in statically typed languages because they're less flexible, so things like that exist for e.g. Java, but not sure if I've seen something for python.

Reply:
Marking as won't fix.